vmware-tanzu/tanzu-framework
# Changes to Addons manager to adopt cluster-api v1beta1

**What this PR does / why we need it**:

* change addons-manager to use cluster-api v1beta1
* bump controller-runtime, controller-tools version to align with cluster-api v1beta1

Signed-off-by: Daniel Guo <danniel1205@gmail.com>

**Which issue(s) this PR fixes**:

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/654

**Describe testing done for PR**:


The following steps are what I have done to verify the changes:
- Build addons-manager image with new changes.
- Create tanzu cluster(1 management cluster and 1 workload cluster) with cluster api v1alpha3.
- Pause `PackageInstall` for addons-manager. (This is to make sure kapp controller will not revert my manual changes back for addons-manager deployment)
- Edit addons-manager deployment to use the new image I built.
- Verified new addons-manager pod is up and running properly.
- Use `clusterclt` to upgrade the management cluster to v1beta1.
- Verified all capi components are running properly.
- Verified addons-manager is still running properly and reconciles packages properly.

**Special notes for your reviewer**:

**Release note**:

```release-note
- addons-manager starts adopting cluster-api v1beta1. (the versions of controller-runtime and controller-tools are bumped to align with cluster-api)
- Rename --metrics-addr to --metrics-bind-addr for addons-manager
- Rename --leader-election to --leader-elect for addons-manager
```
**New PR Checklist**
